### DISCUSSION TITLE
Why does binary_stream_sink::flush not flush the underlying ostream?

### DISCUSSION THREAD

**Discussion by ecorm** [danielaparker/jsoncons#377] [Q&A]
I was comparing `stream_sink` with `binary_stream_sink` to see what the differences were. I noticed that `stream_sink::flush` performs `stream_ptr_->flush()` whereas `binary_stream_sink::flush` does not.

`stream_sink::flush`:
https://github.com/danielaparker/jsoncons/blob/40757fa5dbebcacc79dcfb96bf9485180f13242a/include/jsoncons/sink.hpp#L65-L70

`binary_stream_sink::flush`:
https://github.com/danielaparker/jsoncons/blob/40757fa5dbebcacc79dcfb96bf9485180f13242a/include/jsoncons/sink.hpp#L156-L160

Was this intentional, or was in an omission?

**Comment by danielaparker** [comment]
I can't remember. Off hand I can't think of a reason not to flush it here. It does get flushed when the sink is destroyed.

The only distinguishing feature of `binary_stream_sink` is that the user interacts with the sink using `uint8_t` sequences, and the sink writes to the underlying stream assuming `char`.
